csharp/src/Microsoft.ML.OnnxRuntime/InferenceSession.cs

```diff
@@ -187,12 +190,16 @@ public IDisposableReadOnlyCollection<DisposableNamedOnnxValue> Run(IReadOnlyColl
             }
             finally
             {
-                // always unpin the input buffers, and delete the native Onnx value objects
+                // For NamedOnnxValue, always unpin the input buffers, and delete the native Onnx value objects
+                // For DisposableNamedOnnxValue, the user needs to do this by invoking Dispose
                 for (int i = 0; i < inputs.Count; i++)
                 {
-                    NativeMethods.OrtReleaseValue(inputTensors[i]); // For elementary type Tensors, this should not release the buffer, but should delete the native tensor object.
-                                                                    // For string tensors, this releases the native memory allocated for the tensor, including the buffer
-                    pinnedBufferHandles[i].Dispose();
+                    if (disposeOnnxValues[i])
```

Comment:
When is this disposeOnnxValues ever True? It seems like it's always False -- so this block of code is never executed?

Reply:
It is true for the base class : NamedOnnxValue (thus keeping backwards compatibility). It is false for the derived class (DisposbaleNamedOnnxValue) override method. This is because, by definition of this class, we would want the user to dispose when appropriate and not automatically dispose within the Run().